### PR TITLE
feat: progress bar improvement 

### LINKED ISSUE #129: upgrade GoLang to 1.19
### TL;DR

Update Go to 1.19 everywhere.

### Detailed design

```markdown
Now that go linter supports 1.19 (https://github.com/golangci/golangci-lint/pull/3037),
There's nothing blocking us.
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by gal-legit**
#### What's being changed?
* Add a progress bar for metadata.
* Add a progress bar for secondary rate limits.
* general improvement and minor fixes.
* upgrade Go version to 1.19 for atomic.Bool (fixes #129)

#### Is this PR related to an existing issue?
no
#### Check off the following:

- [x] This PR follows the CONTRIBUTION.md guidelines
- [x] I have self-reviewed my changes before submitting the PR

### CODE REVIEW

**Comment by royb-legit** [comment]
TEST

File: internal/collectors/github/actions_collector.go [hunk 1]
```diff
@@ -57,26 +52,31 @@ func (c *actionCollector) Collect() collectors.SubCollectorChannels {
...
 
-			if err1 != nil || err2 != nil {
-				entityName := fmt.Sprintf("%s/%s", namespace.Organization, org.Name())
-				perm := collectors.NewMissingPermission(permissions.OrgAdmin, entityName, orgActionPermEffect, namespace.Organization)
-				c.IssueMissingPermissions(perm)
-			}
+				if err1 != nil || err2 != nil {
+					entityName := fmt.Sprintf("%s/%s", namespace.Organization, org.Name())
+					perm := collectors.NewMissingPermission(permissions.OrgAdmin, entityName, orgActionPermEffect, namespace.Organization)
+					c.IssueMissingPermissions(perm)
```
Comment (noamd-legit) [inline internal/collectors/github/actions_collector.go:65]: return on error

File: cmd/progressbar/update.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+func NewTimedBar(name string, end time.Time) TimedBarCreation {
+	return TimedBarCreation{
+		BarName: name,
+		End:     end,
+	}
+}
+
+func NewBarClose(name string) BarClose {
```
Comment (noamd-legit) [inline cmd/progressbar/update.go:65]: explain these functions

File: cmd/progressbar/progressbar.go [hunk 3]
```diff
@@ -3,90 +3,183 @@ package progressbar
...
 
-	return p
+	return pb.waiter
+}
+
+func (pb *progressBar) ReportProgress(msg ChannelType) {
+	pb.inChannel <- msg
+}
+
+func (pb *progressBar) handleMinimalBars(data MinimalBars) {
```
Comment (noamd-legit) [inline cmd/progressbar/progressbar.go:100]: minimunRequiredBars

**Review by noamd-legit — APPROVED** [review]

File: internal/clients/github/transport/secondary_rate_limit.go [hunk 4]
```diff
@@ -1,23 +1,43 @@
...
 	"time"
 
+	"github.com/Legit-Labs/legitify/cmd/progressbar"
 	"github.com/gofri/go-github-ratelimit/github_ratelimit"
+	"github.com/gofri/go-github-ratelimit/github_ratelimit/github_ratelimit_test"
 )
 
 const (
 	singleSleepLimit = 90 * time.Second
+	injecterEnvKey   = "LEGITIFY_SECONDARY_RATE_LIMIT_INJECTION"
```
[the remark this answers is not in the document]
Comment (gal-legit) [reply internal/clients/github/transport/secondary_rate_limit.go:17]: It's only useful for testing and will make the code much messier.
It should be removed altogether when we implement the new testing infra with server-mock capabilities, so I thought it'd be better to keep it this way to reduce its fingerprint.
wdyt?

**Review by noamd-legit — APPROVED** [review]

**Pull request merged**